pelican/server.py

```diff
@@ -16,24 +16,30 @@
 
 class ComplexHTTPRequestHandler(srvmod.SimpleHTTPRequestHandler):
     SUFFIXES = ['', '.html', '/index.html']
+    RSTRIP_PATTERNS = ['', '/']
```

Comment:
Sorry for entering this discussion at this late stage, but I wonder whether serving directories should really be prioritised over a html file. I would propose instead:
```
SUFFIXES = ['/index.html', '.html', '']
RSTRIP_PATTERNS = ['/', '']
```
In fact, as is, that final element `'/index.html'` doesn't do anything, since if `os.path.exists(path + '/index.html')`, then also `os.path.exists(path + '')`. The reason why it seems to work now is because `SimpleHTTPRequestHandler.send_head` goes looking for index files if the path in the get request is a directory.

I realise this was introduced by #1040, but it might be opportune to fix that here. If you'd rather merge this first I can open a new issue/pr.

Reply:
This is probably a separate discussion? I'd prefer to get this merged separately, to be honest.

Reply:
Fair enough!